ext/semian/semian_atomic_integer.c

```diff
@@ -0,0 +1,108 @@
+#include "semian_shared_memory_object.h"
+
+typedef struct {
+  int value;
+} semian_atomic_int;
+
+static void semian_atomic_integer_bind_init_fn (size_t byte_size, void *dest, void *prev_data, size_t prev_data_byte_size, int prev_mem_attach_count);
+static VALUE semian_atomic_integer_bind_init_fn_wrapper(VALUE self);
+static VALUE semian_atomic_integer_get_value(VALUE self);
+static VALUE semian_atomic_integer_set_value(VALUE self, VALUE num);
+static VALUE semian_atomic_integer_increase_by(VALUE self, VALUE num);
+
+static void
+semian_atomic_integer_bind_init_fn (size_t byte_size, void *dest, void *prev_data, size_t prev_data_byte_size, int prev_mem_attach_count)
+{
+  semian_atomic_int *ptr = dest;
+  semian_atomic_int *old = prev_data;
+  if (prev_mem_attach_count){
+    if (prev_data){
+      ptr->value = old->value;
+    } // else copy nothing, data is same size and no need to copy
+  } else {
+    ptr->value=0;
+  }
+}
+
+static VALUE
+semian_atomic_integer_bind_init_fn_wrapper(VALUE self)
+{
+  semian_shm_object *ptr;
+  TypedData_Get_Struct(self, semian_shm_object, &semian_shm_object_type, ptr);
+  ptr->object_init_fn = &semian_atomic_integer_bind_init_fn;
+  return self;
+}
+
+static VALUE
+semian_atomic_integer_get_value(VALUE self)
+{
+  semian_shm_object *ptr;
+  TypedData_Get_Struct(self, semian_shm_object, &semian_shm_object_type, ptr);
+
+  // check shared memory for NULL
+  if (0 == ptr->shm_address)
```

Comment:
Handle the possibility of some chain of events leading to the shm not being attached. However, an error of some sort will probably be thrown before then. Should I remove this?